rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/CompactFlowPanel.java

```diff
@@ -0,0 +1,426 @@
+        int index = 0;
+        for (HierarchyNodeSource hierarchyNodeSource : model) {
+            if (hierarchyNodeSource.getNodeSource().getSourceName().equals(host.getSourceName())) {
+                ++index;
+                index += hierarchyNodeSource.getDeploying().size();
+                for (HierarchyHost hierarchyHost : hierarchyNodeSource.getHosts()) {
+                    if (hierarchyHost.getHost().getHostName().equals(host.getHostName())) {
+                        return Optional.of(index);
+                    } else {
+                        index += hierarchyHost.getTiles();
+                    }
+                }
+
+            } else {
+                index += hierarchyNodeSource.getTiles();
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    public Optional<Integer> indexOf(NodeSource.Host.Node node) {
+        int index = 0;
+        for (HierarchyNodeSource hierarchyNodeSource : model) {
+            if (hierarchyNodeSource.getNodeSource().getSourceName().equals(node.getSourceName())) {
+                ++index;
+                if (node.isDeployingNode()) {
+                    for (NodeSource.Host.Node existingNode : hierarchyNodeSource.getDeploying()) {
+                        if (existingNode.equals(node)) {
+                            return Optional.of(index);
+                        } else {
+                            ++index;
+                        }
+                    }
+                } else {
+                    index += hierarchyNodeSource.getDeploying().size();
+                    for (HierarchyHost hierarchyHost : hierarchyNodeSource.getHosts()) {
+                        if (hierarchyHost.getHost().getHostName().equals(node.getHostName())) {
+                            ++index;
+                            for (NodeSource.Host.Node existingNode : hierarchyHost.getNodes()) {
+                                if (existingNode.equals(node)) {
+                                    return Optional.of(index);
+                                } else {
+                                    ++index;
+                                }
+                            }
+                            return Optional.empty();
+                        } else {
+                            index += hierarchyHost.getTiles();
+                        }
+                    }
+
+                }
+                return Optional.empty();
+            } else {
+                index += hierarchyNodeSource.getTiles();
+            }
+        }
+        return Optional.empty();
+    }
+
+}
+
+/**
+ * It is a wrapper around NodeSource
+ * It stores deploying nodes and hosts, in lists so their order it preserved.
+ * Also it has tiles tiles which represent number of tiles which are need to represent
+ * this nodesource with all its content into CompactFlowPanel
+ */
+class HierarchyNodeSource {
+    private NodeSource nodeSource;
+
+    private int tiles = 1;
```

Comment:
I would rename to `numberOfTiles` or `tilesNumber`.

Reply:
Thanks